pycbc/waveform/generator.py

```diff
@@ -1198,57 +1199,57 @@ def generate(self, **kwargs):
 #
 
 
-def select_waveform_generator(approximant):
-    """Returns the single-IFO generator for the approximant.
-
-    Parameters
-    ----------
-    approximant : str
-        Name of waveform approximant. Valid names can be found using
-        ``pycbc.waveform`` methods.
-
-    Returns
-    -------
-    generator : (PyCBC generator instance)
-        A waveform generator object.
-
-    Examples
-    --------
-    Get a list of available approximants:
-    >>> from pycbc import waveform
-    >>> waveform.fd_approximants()
-    >>> waveform.td_approximants()
-    >>> from pycbc.waveform import ringdown
-    >>> ringdown.ringdown_fd_approximants.keys()
-
-    Get generator object:
-    >>> from pycbc.waveform.generator import select_waveform_generator
-    >>> select_waveform_generator(waveform.fd_approximants()[0])
-    """
-    # check if frequency-domain CBC waveform
+def select_waveform_generator(approximant, domain):
+    """Returns the single-IFO generator for the approximant."""
+
+    generator_map = {
+        'fd': {
+            'cbc': FDomainCBCGenerator,
+            'ringdown': {
+                'FdQNMfromFinalMassSpin': FDomainMassSpinRingdownGenerator,
+                'FdQNMfromFreqTau': FDomainFreqTauRingdownGenerator,
+            }
+        },
+        'td': {
+            'cbc': TDomainCBCGenerator,
+            'ringdown': {
+                'TdQNMfromFinalMassSpin': TDomainMassSpinRingdownGenerator,
+                'TdQNMfromFreqTau': TDomainFreqTauRingdownGenerator,
+            },
+            'supernovae': {
+                'CoreCollapseBounce': TDomainSupernovaeGenerator
+            }
+        }
+    }
+
+    gen = None
+
+    # Check if the approximant exists in any domain
     if approximant in waveform.fd_approximants():
-        return FDomainCBCGenerator
-    # check if time-domain CBC waveform
+        if domain in ['fd', 'none']:
+            gen = generator_map['fd']['cbc']
+        elif domain == 'td' and approximant in waveform.td_approximants():
+            gen = generator_map['td']['cbc']
+        elif domain == 'td':
+            raise ValueError(f"{approximant} is a frequency-domain model.")
```

Comment:
Add to the error message "Requested domain is 'td', but...", to give the user a better idea of what went wrong if you get here.